tmgmt_zanata.connector.inc

```diff
@@ -362,9 +367,230 @@ class TmgmtZanataConnector {
     else {
       // TODO include count of failed items.
       $this->job->addMessage(
-        t('One or more items were not successfully sent to Zanata. Try again after correcting the problem items.'),
+        'One or more items were not successfully sent to Zanata. Try again after correcting the problem items.',
         array(), 'error');
     }
+
+    $this->job->addMessage(
+      'Sent existing translations for @sent of @total nodes.',
+        array(
+          '@sent' => $sent_translations_for_how_many,
+          '@total' => count($this->job->getItems())
+        ));
+  }
+
+
+  /**
+   * Check for existing translations to send for a job item, and send them if
+   * they are found.
+   *
+   * @return bool
+   *   TRUE if the translation was submitted successfully, FALSE otherwise
+   */
+  private function sendExistingTranslation(TMGMTJobItem $job_item, $base_options) {
+    // FIXME this check may not be necessary, but keep it for now until it is
+    //       clear what an item type can be other than node.
+    $is_node = strcmp($job_item->item_type, 'node') === 0;
+    $id = $job_item->item_id;
+
+    if ($is_node) {
+      $trans_data = $this->getExistingTransData($job_item);
+      if ($trans_data !== NULL) {
+        $url = $this->getItemTranslationUrl($job_item);
+
+        $data = array(
+          'textFlowTargets' => $this->prepareTargets($trans_data, $job_item),
+        );
+
+        // makes a copy
+        $options = $base_options;
+        $options['data'] = json_encode($data);
+        $response = drupal_http_request($url, $options);
+
+        switch ($response->code) {
+          case 200:
+          case 201:
+            // translations were successfully sent
+            return TRUE;
+
+          case 400:
+            drupal_set_message(
+              t('There was a problem with the request to send existing translations: "@error"',
+                array('@error' => $response->error)),
+              'error',
+              FALSE);
+            return FALSE;
+
+          case 404:
+            $this->showDocumentNotFoundErrorMessage();
+            return FALSE;
+
+          case 401:
+            $this->showAuthErrorMessage();
+            return FALSE;
+
+          case 500:
+          default:
+            drupal_set_message(
+              t('There was a problem sending translations to Zanata. See messages below.'),
+              'warning',
+              FALSE);
+            $job_item->addMessage(
+              'Tried to send existing translations but got response code @code when sending to URL @url.',
+              array(
+                '@code' => $response->code,
+                '@url' => $url));
+            return FALSE;
+        }
+      }
+      else {
+        // no existing translations found. Probably don't need a log message.
+      }
+
+    }
+    else {
```

Comment:
Good suggestion, I will do that.
- [ ] change `else` portion to a check that returns if failed.